Homework/Truong_Tran/lesson_6/MinimumGeneticMutation.java

```diff
@@ -0,0 +1,129 @@
+import java.util.*;
+
+public class MinimumGeneticMutation {
+
+    public int minMutation(String startGene, String endGene, String[] bank) {
+        if (startGene.length() != endGene.length()) {
+            return -1;
+        }
+        Map<String, Boolean> visited = new HashMap<>();
+        for (int i = 0; i < bank.length; i++) {
+            visited.put(bank[i], false);
+        }
+        List<Integer> indexs = preProcessing(startGene, endGene);
+        Queue<String> queue = new ArrayDeque<>();
+        queue.add(startGene);
+        int depth = 0;
+        while (!queue.isEmpty()) {
+            List<String> next = new ArrayList<>();
+            while (!queue.isEmpty()) {
+                String gene = queue.poll();
+                if (endGene.equals(gene)) {
+                    return depth;
+                }
+                for (Integer i : indexs) {
+                    Character c = gene.charAt(i);
+                    switch (c) {
+                        case 'A':
```

Comment:
repeated logic, i don't think you have enough time to type this much code in a real interview. 
can improve? I see Trong has shorter logic https://github.com/quanghoang33/Gen-7/pull/17/files#diff-9e570e0ca9da09266c9b821a3177b409a8281c942185dc7d1c702293506e64f3